File: split/initrow.pas
procedure initrow;
begin
  pushnest;
  curlist.modefield := (-103) - curlist.modefield;
  if curlist.modefield = -102 then
    curlist.auxfield.hh.lh := 0
  else
    curlist.
      auxfield.int := 0;
  begin
    mem[curlist.tailfield].hh.rh := newglue(mem[mem[29992].hh.rh + 1].hh.lh);
    curlist.tailfield := mem[curlist.tailfield].hh.rh;
  end;
  mem[curlist.tailfield].hh.b1 := 12;
  curalign := mem[mem[29992].hh.rh].hh.rh;
  curtail := curhead;
  initspan(curalign);
end;

File: cpp/src/initrow.cpp
#include "initrow.h"
#include "pushnest.h"
#include "initspan.h"
#include "noeud.h"

void initrow(void)
{
	pushnest();
	mode = (-hmode-vmode)-mode;
	if (mode == -hmode)
		space_factor = 0;
	else
		prev_depth = 0;
	auto G = new GlueNode(dynamic_cast<GlueNode*>(preamble)->glue_ptr);
	G->subtype = tab_skip_code+1;
	tail_append(G);
	curalign = dynamic_cast<AlignRecordNode*>(preamble->link);
	curtail = curhead;
	initspan(dynamic_cast<SpanNode*>(curalign));
}
